hbase-client/src/main/java/org/apache/hadoop/hbase/filter/PrefixFilter.java

```diff
@@ -69,13 +74,19 @@ public boolean filterRowKey(Cell firstRowCell) {
     if ((!isReversed() && cmp > 0) || (isReversed() && cmp < 0)) {
       passedPrefix = true;
     }
-    filterRow = (cmp != 0);
+    filterRow = (!isReversed() && cmp > 0) || (isReversed() && cmp < 0);
```

Comment:
This is the same as passedPrefix above. No need to compute twice.

Reply:
Thanks, good point. Actually I was also thinking about this but left it eventually. I'll remove computing this again.

Reply:
Should be fixed now.